virtcontainers/device/manager/manager.go

```diff
@@ -229,3 +230,41 @@ func (dm *deviceManager) IsDeviceAttached(id string) bool {
 	}
 	return d.GetAttachCount() > 0
 }
+
+// NewDevice creates a device based on specified DeviceInfo
+func (dm *deviceManager) LoadDevices(devStates []persistapi.DeviceState) {
+	dm.Lock()
+	defer dm.Unlock()
+
+	for _, ds := range devStates {
+		switch config.DeviceType(ds.Type) {
+		case config.DeviceGeneric:
+			dev := &drivers.GenericDevice{}
+			dev.Load(ds)
+			dm.devices[dev.DeviceID()] = dev
+		case config.DeviceBlock:
+			dev := &drivers.BlockDevice{}
+			dev.Load(ds)
```

Comment:
Nit: It would be great if we could avoid all the `dev.Load()` and `dm.devices[dev.DeviceID()]` duplication somehow  (an interface?) so that those calls could be made once, after the `switch`

Reply:
Sounds good.